gh-2100 - GetVariable and SetVariable operations don't work properly with FederatedStore

c015dariu opened the pull request:
#2100 

One line change within Context constructor that is called via shallowClone()

m55624 commented:
Sorry @c015dariu, just added a comment to the ticket.  Shouldn't be long.

c015dariu commented:
@m55624 added the test, had to change the context builder because despite allowing the addition of variables, .build() did not actually apply them to the created context. The constructor used in .build() is deprecated however I've left it as it is however with the addition of a small check to see whether to add variables or not.

m55624 commented:
Ah good spot.  Even deprecated it should have been added to the builder correctly.  Cheers.

c015dariu commented:
@m55624 Just noticed another bug within context regarding addVariables() where it is accidentally referencing the passed in variable for null rather than whether a map currently exists within the object which means it could crash if a variable is added without using set first. Just working on fixing that one and making sure it hasn't broken anything now.

m55624 commented:
Ah yeah, it's missing the "this".  Cheers for that.

c015dariu commented:
Okay, so I decided to add a couple of tests to cover the variables side of Context and came across an issue where Collections.singletonMap was being used to instantiate variables where none existed, this meant context could only hold one variable in the map and would throw an exception if any others were added. I've now added two tests to cover setVariables and addVariables (previously unused) and fixed the two methods. Uploading once checkstyle and tests have passed on my end.

m55624 commented on core/store/src/test/java/uk/gov/gchq/gaffer/store/ContextTest.java:
```diff
@@ -141,4 +143,65 @@ public void shouldSetAndGetOriginalOpChain() {
+        context.setVariable("testVar2", "testVarVal2");
+
+        // Then
+        assertTrue(!context.getVariables().isEmpty());
```
can we change this to assertFalse and remove the not operator.  Just can be slightly confusing to skip read if you miss the not.

m55624 commented on core/store/src/test/java/uk/gov/gchq/gaffer/store/ContextTest.java:
```diff
@@ -141,4 +143,65 @@ public void shouldSetAndGetOriginalOpChain() {
+        context.addVariables(Collections.singletonMap("testVar2", "testVarVal2"));
+
+        // Then
+        assertTrue(!context.getVariables().isEmpty());
```
Same here

c015dariu commented:
Requested changes resolved.
